### ISSUE TITLE
Module installation with Drupal 11

### ISSUE THREAD

**Issue by mscieszka** [ddev/ddev-drupal-contrib#38]
Currently, the Drupal version to install is determined in [expand-composer-json](https://github.com/ddev/ddev-drupal-contrib/blob/b3d5dc63bf434501af0989547bc1e7d710040041/commands/web/expand-composer-json#L12) based on `$DDEV_PROJECT_TYPE`.

It looks like the current approach won't work for Drupal 11 when [DDEV introduces D11 support](https://github.com/ddev/ddev/pull/5991), as `$DDEV_PROJECT_TYPE` won't have a `drupal11` value, but will simply be `drupal`.

**Comment by rfay** [comment]
DDEV does detect and use the Drupal version though. Should it be introduced as a new variable or something?

**Comment by mscieszka** [comment]
If a contrib module supports (or aims to support) both D10 and D11, what would be the [installation flow](https://github.com/ddev/ddev-drupal-contrib?tab=readme-ov-file#install) to explicitly install Drupal 11 instead of a Drupal 10?

To achieve this now, I have manually added `[[ $DDEV_PROJECT_TYPE == "drupal" ]] && export _TARGET_CORE=11.x-dev` to the expand-composer-json file - I am just not sure if there is a better way than this "hack".

**Comment by mscieszka** [comment]
There is another approach that does not involve modifying composer.json:
For example, in `.ddev/commands/web/poser', just before `composer install', we can add
`composer require drupal/core-recommended:11.x-dev@dev drupal/core-composer-scaffold:11.x-dev@dev drupal/core-project-message:11.x-dev@dev --update-with-all-dependencies` to install Drupal 11.x-dev.

As a side note, I was getting the error 'error This project's package.json defines "packageManager": "yarn@4.1.1". However, the current global version of Yarn is 1.22.22'. - To fix this, I also added `yarn policies set-version stable` to the poser file.

**Comment by Wylbur** [comment]
@mscieszka Thanks for you quick work!  I applied your fix to _.ddev/commands/web/poser_, but I had to add the latest version of drush to the composer require line.  

Here's the edited poser file.  

```export COMPOSER=composer.contrib.json
.ddev/commands/web/expand-composer-json "$DDEV_PROJECT_NAME"
composer require drupal/core-recommended:11.x-dev@dev drupal/core-composer-scaffold:11.x-dev@dev drupal/core-project-message:11.x-dev@dev drush/drush:13.x-dev --update-with-all-dependencies
composer install
rm composer.contrib.json composer.contrib.lock
yarn policies set-version stable
yarn --cwd $DDEV_DOCROOT/core install
touch $DDEV_DOCROOT/core/.env```

**Comment by weitzman** [comment]
I took a different direction than #45. See my PR in #46. Feedback welcome.

**Comment by weitzman** [comment]
Fixed in #46
